src/pykiso/lib/connectors/cc_pcan_can.py

```diff
@@ -369,3 +378,69 @@ def _cc_receive(
         except Exception:
             log.exception(f"encountered error while receiving message via {self}")
             return {"msg": None}
+
+    def _merge_trc(self) -> None:
+        """Merge multiple trc files in one."""
+        list_of_traces = []
+
+        if isinstance(self.trace_path, str):
+            self.trace_path = Path(self.trace_path)
+
+        # Get all trc files in trace path
+        list_of_all_traces = list(self.trace_path.glob("*.trc"))
+
+        log.internal_debug(f"merging following trace files: {list_of_all_traces}")
+        # Select a number of the lastest trace files corresponding to the number of traces created.
+        for _ in range(self.trc_count):
+            try:
+                latest_trace = max(list_of_all_traces, key=os.path.getctime)
+                list_of_all_traces.pop(list_of_all_traces.index(latest_trace))
+                list_of_traces.append(latest_trace)
+            except ValueError:
+                break
+
+        list_of_traces.reverse()
+        try:
+            # If name not provided, take the fist trace created as results file
+            if self.trace_name is None:
+                result_trace = list_of_traces[0]
+            else:
+                result_trace = str(self.trace_path / self.trace_name)
+
+            if self.trace_name is not None:
+                with open(list_of_traces[0], "r") as trc:
+                    merged_data = trc.read()
+                with open(result_trace, "w") as merged_trc:
+                    merged_trc.write(merged_data)
+                os.remove(list_of_traces[0])
+
+            list_of_traces.pop(0)
+            # End of the trace header
+            trc_start = 33
+
+            for file in list_of_traces:
+                with open(file, "r") as trc:
+                    data = trc.read().splitlines(True)
+                with open(result_trace, "a") as merged_trc:
+                    merged_trc.writelines(data[trc_start:])
+                os.remove(file)
+
+            with open(result_trace, "r+") as trc:
+                merged_data_lines = trc.read().splitlines(True)
+
+                # Parsing to keep the message numbers consistent after the merge
+                for line_number in range(trc_start, len(merged_data_lines)):
```

Comment:
this 3 lines seems to be a little bit of magic with line_number + 1 rjust 7 and [7:]

Reply:
Should be better now